app/experimenter/experiments/serializers/design.py

```diff
@@ -321,14 +321,18 @@ def update(self, instance, validated_data):
             existing_preference_ids = set(
                 instance.preferences.all().values_list("id", flat=True)
             )
+
+            submitted_preference_ids = []
             for preference_data in preferences_data:
                 preference_data["experiment"] = instance
-                RolloutPreference.objects.create(**preference_data)
+                pref_id = preference_data.pop("id", None)
+                if pref_id:
+                    submitted_preference_ids.append(pref_id)
+                RolloutPreference.objects.update_or_create(
+                    id=pref_id, defaults=preference_data
```

Comment:
Oh okay good so this worked, we should try to use this pattern everywhere we can I think.